### DISCUSSION TITLE
Does Tempo supports AWS S3 encryption?

### DISCUSSION THREAD

**Discussion by slenky** [grafana/tempo#1301] [Q&A]
Hello,

I would like everything stored in S3 to be encrypted either by AES256 or AWS KMS. I have not found any configuration for Tempo to support encryption. Could you please tell me if it's actually supported?

Regards,
Bohdan

**Comment by joe-elliott** [comment]
Our current config is here and does not support encryption:

https://github.com/grafana/tempo/blob/main/tempodb/backend/s3/config.go

We are using the [minio client](https://github.com/minio/minio-go) so it wouldn't be too difficult to expose any features that already exist there. 

Is this something that can be configured transparently to the client? Directly on the bucket? or does the client need to be aware as well?

**Comment by zalegrala** [comment]
There is also https://github.com/grafana/tempo/issues/1486 currently open.
